File: VisualBasicProject/VisualBasicProject.MsTestFw/StepDefinitions/CalculatorStepDefinitions.vb
Imports Microsoft.VisualStudio.TestTools.UnitTesting
Imports Reqnroll

Namespace StepDefinitions

    <Binding>
    Public Class CalculatorStepDefinitions

        Private ReadOnly _testContext As TestContext

        Public Sub New(testContext As TestContext)
            _testContext = testContext
        End Sub

        <Given("the first number is {int}")>
        Public Sub GivenTheFirstNumberIs(number As Integer)
        End Sub

        <Given("the second number is {int}")>
        Public Sub GivenTheSecondNumberIs(number As Integer)
        End Sub

        <Reqnroll.When("the two numbers are added")>
        Public Sub WhenTheTwoNumbersAreAdded()
            Console.WriteLine(_testContext.CurrentTestOutcome)
        End Sub

        <Reqnroll.Then("the result should be {int}")>
        Public Sub ThenTheResultShouldBe(result As Integer)
            Assert.AreEqual(result, result)
        End Sub
    
    End Class
End NameSpace

File: CleanReqnrollProject/CleanReqnrollProject.Net8.MsTest4Fw/StepDefinitions/CalculatorStepDefinitions.cs
namespace CleanReqnrollProject.StepDefinitions;

[Binding]
public sealed class CalculatorStepDefinitions
{
    private readonly TestContext testContext;

    public CalculatorStepDefinitions(TestContext testContext)
    {
        this.testContext = testContext;
    }

    [Given("the first number is {int}")]
    public void GivenTheFirstNumberIs(int number)
    {
    }

    [Given("the second number is {int}")]
    public void GivenTheSecondNumberIs(int number)
    {
    }

    [When("the two numbers are added")]
    public void WhenTheTwoNumbersAreAdded()
    {
        Console.WriteLine(testContext.CurrentTestOutcome);
    }

    [Then("the result should be {int}")]
    public void ThenTheResultShouldBe(int result)
    {
        Assert.AreEqual(result, result);
    }
}
